src/encodings/rle.rs

```diff
@@ -189,6 +189,13 @@ impl RleEncoder {
     self.bit_writer.consume()
   }
 
+  /// Borrow equivalent of the `consume` method.
+  /// Call `clear()` after invoking this method.
+  #[inline]
+  pub fn flush_buffer(&mut self) -> &[u8] {
+    self.bit_writer.flush_buffer()
```

Comment:
Sorry, I was thinking that this can also call `flush` first:

```rust
pub fn flush_buffer(&mut self) -> Result<&[u8]> {
  self.flush()?
  self.bit_writer.flush_buffer()
}
```
So that it matches the behavior of other `flush_buffer()`s (otherwise, have to call `flush()` before `flush_buffer()` is a little confusing.) However this makes `flush_buffer()` a little hard to change though because of lifetime error. I made it work in this way:
```rust
  #[inline]
  fn flush_buffer(&mut self) -> Result<ByteBufferPtr> {
    assert!(self.encoder.is_some(), "RLE value encoder is not initialized");
    let rle_encoder = self.encoder.as_mut().unwrap();

    // Flush all encoder buffers and raw values
    let encoded_data = {
      let buf = rle_encoder.flush_buffer()?;

      let len = (buf.len() as i32).to_le();
      let len_bytes = len.as_bytes();
      let mut encoded_data = Vec::new();
      encoded_data.extend_from_slice(len_bytes);
      encoded_data.extend_from_slice(buf);
      encoded_data
    };

    // Reset rle encoder for the next batch
    rle_encoder.clear();

    Ok(ByteBufferPtr::new(encoded_data))
  }
```

Also, `test_rle_consume_flush_buffer` needs to call `flush()` on `encoder2` before `consume()`.

Let me know what you think.

Reply:
All good. Looks interesting. I will try it soon, thanks.

Reply:
Do we really need to make this change? I was thinking that this might force us to change `consume()` method to behave the same way. Maybe we should simply rename `flush_buffer` to something like `get_buffer`? Or should I proceed with this change?

Reply:
Hmm. Good point. IMO we should change `consume()` too - just like `bit_util::flush_buffer()`, `bit_util::consume()` also flushes before consuming it. So, the `rle::consume()` and `rle::flush_buffer()` should just directly call the corresponding ones in `bit_util`. Let me know what you think.